sdk/search/azure-search/src/test/java/com/azure/search/DocumentConverterTests.java

```diff
@@ -0,0 +1,292 @@
+                                        put("city", "Vancouver");
+                                        put("country", "Canada");
+                                    }
+                                }
+                            );
+                            put("location", new Document() {
+                                    {
+                                        put("type", "Point");
+                                        put("coordinates", Arrays.asList(-121, 49));
+                                    }
+                                }
+                            );
+                        }
+                    },
+                    new Document() {
+                        {
+                            put("name", "South");
+                            put("address", new Document() {
+                                {
+                                    put("city", "Seattle");
+                                    put("country", "USA");
+                                }
+                            });
+                            put("location", new Document() {
+                                {
+                                    put("type", "Point");
+                                    put("coordinates", Arrays.asList(-122.5, 47.6));
+                                }
+                            });
+                        }
+                    }
+                ));
+            }
+        };
+
+        Document actualDoc = deserialize(json);
+        Assert.assertEquals(expectedDoc, actualDoc);
+    }
+
+    @Test
+    public void dateTimeStringsAreReadAsDateTime() {
+        String json = "{\"field1\":\"".concat(testDateString).concat("\",\"field2\" : [\"").concat(testDateString).concat("\", \"").concat(testDateString).concat("\"]}");
+        Document expectedDoc = new Document() {
+            {
+                put("field1", testDate);
+                put("field2", Arrays.asList(testDate, testDate));
+            }
+        };
+
+        Document actualDoc = deserialize(json);
+        Assert.assertEquals(expectedDoc, actualDoc);
+    }
+
+    @Test
+    public void specialDoublesAreReadAsStrings() {
+        String json = "{\"field1\" : \"NaN\", \"field2\": \"INF\", \"field3\": \"-INF\", \"field4\": [\"NaN\", \"INF\", \"-INF\"], \"field5\": {\"value\":\"-INF\"}}";
+        Document expectedDoc = new Document() {
+            {
+                put("field1", "NaN");
+                put("field2", "INF");
+                put("field3", "-INF");
+                put("field4", Arrays.asList("NaN", "INF", "-INF"));
+                put("field5", new Document() {
+                    {
+                        put("value", "-INF");
+                    }
+                });
+            }
+        };
+
+        Document actualDoc = deserialize(json);
+        Assert.assertEquals(expectedDoc, actualDoc);
+    }
+
+    @Test
+    public void dateTimeStringsInArraysAreReadAsDateTime() {
```

Comment:
Dur to the decision to park the custom desialization of documents to typed classes and instead return a map of objects, not all custom tests got implemeted and were defered.

Reply:
These tests are not related to the deserialization to typed classes. We'll have to port them before we ship; Otherwise we don't know how the SDK will handle these cases.

We can do that in a separate PR.